pynamodb/attribute_dict.py

```diff
@@ -0,0 +1,33 @@
+import collections
+
+
+class AttributeDict(collections.MutableMapping):
+    """
+    A dictionary that stores attributes by two keys
+    """
+    def __init__(self, *args, **kwargs):
+        self._values = {}
+        self._alt_values = {}
```

Comment:
the names here are a bit unclear, one is basically for the key used by dynamo, the other is the key used by the object right?

I think it's a bit more clear if this was explicit instead why not just use two dictionaries? If the keys are the same you can store the same on both